dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/Ds3PanelPresenter.java

```diff
@@ -322,8 +325,28 @@ private void createTabAndSetBehaviour(final Session newSession) {
 
         });
         treeTab.setOnCloseRequest(SafeHandler.logHandle(event -> {
-            ds3SessionStore.removeSession(getSession());
-            closeTab((Tab) event.getSource(), getSession());
+            final ImmutableList<Ds3JobTask> notCachedRunningTasks = jobWorkers.getTasks().stream()
+                    .filter(ds3JobTask -> CancelJobsWorker.compareEndpoints(getSession(), ds3JobTask))
+                    .filter(task -> task.getProgress() != 1).collect(GuavaCollectors.immutableList());
```

Comment:
Pull this out into a method to state what the check is looking for, ie. isJobFullyInCache